autogen/agentchat/contrib/capabilities/context_handling.py

```diff
@@ -0,0 +1,112 @@
+import sys
+from termcolor import colored
+from typing import Dict, Optional, Union, List, Tuple, Any
+from autogen.agentchat.contrib.capabilities.agent_capability import AgentCapability
+from autogen import ConversableAgent
+from autogen import token_count_utils
+
+
+class TransformChatHistory:
+    """
+    An agent's chat history with other agents is a common context that it uses to generate a reply.
+    This capability allows the agent to transform its chat history prior to using it to generate a reply.
+    It does not permanently modify the chat history, but rather processes it on every invocation.
+
+    """
```

Comment:
This docstring is going to be what user will see so we shall add the doc string from the private method `_transform_messages` to here.